extensions/resource-deployment/src/ui/modelViewUtils.ts

```diff
@@ -316,6 +324,46 @@ export function initializeWizardPage(context: WizardPageContext): void {
 	});
 }
 
+/**
+ * Hooks up the dynamic enablement for fields which use that. This will attach a listener to the target component
+ * for when the value changes and update the enabled state of the source component based on the current value
+ * of the target component.
+ *
+ * Note that currently this is only supported for Notebook Wizard Pages and only supports direct equals comparison
+ * for the value currently selected.
+ *
+ * Additionally this only supports hooking up components that are on the same page.
+ * @param context The page context
+ */
+function hookUpDynamicEnablement(context: WizardPageContext): void {
+	context.pageInfo.sections.forEach(section => {
+		section.fields?.forEach(field => {
+			if (instanceOfDynamicEnablementInfo(field.enabled)) {
+				const fieldKey = field.variableName || field.label;
+				const fieldComponent = context.inputComponents[fieldKey];
+				const targetComponent = context.inputComponents[field.enabled.target];
+				const targetValue = field.enabled.value;
+				if (!targetComponent) {
+					console.error(`Could not find target component ${field.enabled.target} when hooking up dynamic enablement for ${field.label}`);
+					return;
+				}
+				targetComponent.onValueChanged(() => {
+					fieldComponent.component.enabled = targetComponent.getValue() === targetValue;
+					// We also need to update the required flag so that when the component is disabled it won't block the page from proceeding
+					if ('required' in fieldComponent.component) {
+						fieldComponent.component.required = fieldComponent.component.enabled === false ? false : field.required;
+					}
+				});
+				// Initialize the state based on the current value of the component
+				fieldComponent.component.enabled = targetComponent.getValue() === targetValue;
```

Comment:
have an method to set the enabled and required properties? this is repeated twice.

Reply:
Eh, for four lines that are right next to each other I'm not concerned about it. That's just adding more text for not a whole lot of gain.

Reply:
Creating in the an anonymous funciton within the body of this function for this is still useful. The point is to make a statement that set the field value using some code logic and set the field value on targetComponent's valueChanged using the same logic.

Reply:
Done.